jwk/helper.go

```diff
@@ -120,6 +127,26 @@ func FindKeyByPrefix(set *jose.JSONWebKeySet, prefix string) (key *jose.JSONWebK
 	return First(keys.Keys), nil
 }
 
+func FindPublicKey(set *jose.JSONWebKeySet) (key *jose.JSONWebKey, err error) {
+	keys := ExcludePrivateKeys(set)
+	if len(keys.Keys) == 0 {
+		return nil, errors.New("key not found")
+	}
+
+	return First(keys.Keys), nil
+}
+
+func ExcludePrivateKeys(set *jose.JSONWebKeySet) *jose.JSONWebKeySet {
+	keys := new(jose.JSONWebKeySet)
+	for _, k := range set.Keys {
+		_, ecdsaOk := k.Key.(*ecdsa.PrivateKey)
+		if _, ok := k.Key.(*rsa.PrivateKey); !ok && !ecdsaOk {
+			keys.Keys = append(keys.Keys, k)
+		}
+	}
+	return keys
```

Comment:
This list should not filter private keys out, it should only keep public keys in :)

Reply:
It's a bit unclear for me what I should change in this function. 
Should the function delete the private keys from the existing list instead of copying the public keys into another list?

Reply:
Something like this ? 

`func ExcludePrivateKeys(set *jose.JSONWebKeySet) *jose.JSONWebKeySet {
	keys := new(jose.JSONWebKeySet)
	i := 0 // output index
	for _, k := range set.Keys {
		_, ecdsaOk := k.Key.(*ecdsa.PublicKey)
		_, dsaOK := k.Key.(*dsa.PublicKey)
		_, ed25519OK := k.Key.(*ed25519.PublicKey)
		_, rsaOK := k.Key.(*rsa.PublicKey)

		if !(ecdsaOk || dsaOK || ed25519OK || rsaOK) {
			set.Keys[i] = k
			i++
		}
	}
	return keys
}`

Reply:
Exactly! Sorry for the late reply.